gef-ui/server/controllers/application-details/index.test.js

```diff
@@ -352,27 +350,58 @@ describe('controllers/application-details', () => {
       it('renders `review-decision` when page requested is `review-decision`', async () => {
         mockApplicationResponse.status = CONSTANTS.DEAL_STATUS.UKEF_APPROVED_WITHOUT_CONDITIONS;
         api.getApplication.mockResolvedValueOnce(mockApplicationResponse);
+        const req = {
+          ...MOCKS.MockRequestUrl('/gef/application/123/review-decision'),
+          flash: jest.fn().mockReturnValue('Facility is updated'),
+        };
+        await applicationDetails(req, mockResponse);
 
-        await applicationDetails(MOCKS.MockRequestUrl('/gef/application/123/review-decision'), mockResponse);
+        expect(mockResponse.render)
+          .toHaveBeenCalledWith('partials/review-decision.njk', expect.objectContaining({ applicationStatus: mockApplicationResponse.status }));
+        expect(req.flash).toHaveBeenCalledWith('success');
+      });
+
+      it('renders `unissued-facilities` when page requested is `unissued facilities` ', async () => {
+        mockApplicationResponse.status = CONSTANTS.DEAL_STATUS.UKEF_ACKNOWLEDGED;
+        api.getApplication.mockResolvedValueOnce(mockApplicationResponse);
+        const req = {
+          ...MOCKS.MockRequestUrl('/gef/application/123/unissued-facilities'),
+          flash: jest.fn().mockReturnValue('Facility is updated'),
+        };
+        await applicationDetails(req, mockResponse);
 
         expect(mockResponse.render)
-          .toHaveBeenCalledWith('partials/review-decision.njk', expect.objectContaining({
+          .toHaveBeenCalledWith('partials/unissued-facilities.njk', expect.objectContaining({
             applicationStatus: mockApplicationResponse.status,
+            unissuedFacilitiesPresent: false,
+            facilitiesChangedToIssued: [],
           }));
+        expect(req.flash).toHaveBeenCalledWith('success');
       });
 
-      it('renders `unissued-facilities` when page requested is `unissued facilities` ', async () => {
+      it('to check if flash: jest.fn().mockReturnValue does not exist', async () => {
         mockApplicationResponse.status = CONSTANTS.DEAL_STATUS.UKEF_ACKNOWLEDGED;
         api.getApplication.mockResolvedValueOnce(mockApplicationResponse);
-
-        await applicationDetails(MOCKS.MockRequestUrl('/gef/application/123/unissued-facilities'), mockResponse);
+        const req = {
+          ...MOCKS.MockRequestUrl('/gef/application/123/unissued-facilities'),
+          flash: jest.fn().mockReturnValue('Facility is updated'),
+        };
+        await applicationDetails(req, mockResponse);
 
         expect(mockResponse.render)
           .toHaveBeenCalledWith('partials/unissued-facilities.njk', expect.objectContaining({
             applicationStatus: mockApplicationResponse.status,
             unissuedFacilitiesPresent: false,
             facilitiesChangedToIssued: [],
           }));
+        req.flash.mockReset();
+        expect(req.flash).not.toHaveBeenCalled();
```

Comment:
We should not be resetting a mock prior to checking to see how much it has been called -- I think this will give a false `0`. This is evident in that it should always be called on line 257 of the `index.js` file

Reply:
Removed `mockReset()`

Reply:
resolved as this is no longer needed